### PR TITLE
Use scriptlet to get current version for snapcraft.yaml

### LINKED ISSUE #1735: Integrate snapcraft.yaml with cmake configure_file()
**Is your feature request related to a problem? Please describe.**
New `snapcraft.yaml` file is hard coded to a version tag. We want to have the file use the cmake `configure_file()` function going forward.

**Describe the solution you'd like**
Rename `snapcraft.yaml` as `snapcraft.yaml.in` and then update our `CMakeLists.txt` file to configure it correctly in the `build` directory.

**Additional context**
Right now, `snapcraft.yaml` will only work on the `stable` branch. We need to wait until the snap team gets Qt support for Ubuntu 22.04 via `core22`. We have a higher cmake version requirement in `master` branch that breaks the snap capability.

### PR DESCRIPTION
**Pull request by jwrober**
Closes #1735

Part of this might be good for `stable`

### CODE REVIEW

File: dist/snapcraft.yaml [hunk 1]
```diff
@@ -48,10 +46,13 @@ icon: snap/local/freeciv21-client.png
 
 parts:
   freeciv21:
+    override-pull: |
+      snapcraftctl pull
+      snapcraftctl set-version `sed -n '2p' snap/local/AutoRevision.txt`
+      snapcraftctl set-grade "devel"
```
Comment (lmoureaux) [inline dist/snapcraft.yaml:52]: Shouldn't this also be computed dynamically?

Comment (jwrober) [reply dist/snapcraft.yaml:52]: Suggestions on how to do that?

Comment (lmoureaux) [reply dist/snapcraft.yaml:52]: The three possible values are `devel`, `candidate`, and `stable`. What about setting `candidate` for rc releases and `stable` for stable releases?

Comment (jwrober) [reply dist/snapcraft.yaml:52]: I guess I could write an `if` statement that uses `awk` or something to tease out part of the version. `alpah` then gives `devel`, `rc` gives `candidate` and finally the absence of those gives `stable`.

**Comment by lmoureaux** [comment]
Anything I can help with to move this PR forward?

**Comment by jwrober** [comment]
I was looking at it yesterday after reviewing your other PR. I will have some time this week to poke on it. Any suggestions you may have are always welcome.

**Review by lmoureaux — APPROVED** [review]

**Comment by lmoureaux** [comment]
> Part of this might be good for stable

We can add to the backport list if you think this will rebase fine. Else we need a second PR

**Pull request merged**